pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -472,91 +472,124 @@ func ResolvePipelineRunTask(
 	getTaskRun resources.GetTaskRun,
 	getRun GetRun,
 	getCondition GetCondition,
-	task v1beta1.PipelineTask,
+	pipelineTask v1beta1.PipelineTask,
 	providedResources map[string]*resourcev1alpha1.PipelineResource,
 ) (*ResolvedPipelineRunTask, error) {
-
 	rprt := ResolvedPipelineRunTask{
-		PipelineTask: &task,
+		PipelineTask: &pipelineTask,
 	}
 	rprt.CustomTask = isCustomTask(ctx, rprt)
 	if rprt.IsCustomTask() {
-		rprt.RunName = getRunName(pipelineRun.Status.Runs, pipelineRun.Status.ChildReferences, task.Name, pipelineRun.Name)
+		rprt.RunName = getRunName(pipelineRun.Status.Runs, pipelineRun.Status.ChildReferences, pipelineTask.Name, pipelineRun.Name)
 		run, err := getRun(rprt.RunName)
 		if err != nil && !kerrors.IsNotFound(err) {
 			return nil, fmt.Errorf("error retrieving Run %s: %w", rprt.RunName, err)
 		}
 		rprt.Run = run
 	} else {
-		rprt.TaskRunName = GetTaskRunName(pipelineRun.Status.TaskRuns, pipelineRun.Status.ChildReferences, task.Name, pipelineRun.Name)
-
-		taskRun, err := getTaskRun(rprt.TaskRunName)
-		if err != nil {
-			if !kerrors.IsNotFound(err) {
-				return nil, fmt.Errorf("error retrieving TaskRun %s: %w", rprt.TaskRunName, err)
-			}
-		}
-		if taskRun != nil {
-			rprt.TaskRun = taskRun
+		rprt.TaskRunName = GetTaskRunName(pipelineRun.Status.TaskRuns, pipelineRun.Status.ChildReferences, pipelineTask.Name, pipelineRun.Name)
+		if err := resolvePipelineRunTaskWithTaskRun(ctx, &rprt, rprt.TaskRunName, pipelineRun, getTask, getTaskRun, getCondition, pipelineTask, providedResources); err != nil {
```

Comment:
is it possible to retrieve `taskRunName` from `rprt`? If so, we can avoid including `taskRunName` from the list of parameters?

Reply:
discussed on slack, but adding comment here for future reference:

yes, but we need `taskRunName` to be its own field separate from the rest of the `rprt` is that it sets us up to support matrixed `rprt` which will have list of `taskRunNames` to iterate over